### ISSUE TITLE
[Bug]: Incorrectly reporting media downloaded by different user

### ISSUE THREAD

**Issue by ljlongwing** [advplyr/audiobookshelf-app#1320]
### What was the Problem?

I log onto my server and always download the books locally while listening so that I'm not reliant on connection to the server, the download typically works well.  But recently I've been getting red ! marks on the image, with a note that states "Media is linked to this server but was downloaded by a different user.  Progress will only be synced to the user that downloaded it."  I only connect with one user, so it's entirely not possible to be a different user

### Steps to Reproduce the Issue

I'm not getting this all the time, but frequent, so I'm not sure about reproduction steps

### What was Expected?

it would be nice if it recognized that I'm me and I'm the one that downloaded the book

### Phone Model

Samsung Galaxy S22+

### Phone OS

Android 14

### Audiobookshelf App Version

Android App - 0.9.76

### Installation Source

Google Play Store

**Comment by nichwall** [comment]
There have been a few reports of this in Discord. Did you update your server after downloading the books?

Does resetting the app data and logging back in to the server fix it?

**Comment by ljlongwing** [comment]
Yes, I keep my server updated, and this happens regularly, while I might have the books on my phone for awhile, so it's entirely probable that the server has been upgraded between downloads.

So, resetting the app caused it to not know about my downloads folder, which of course made it not know the book was already downloaded....re-adding my download folder doesn't seem to have made it aware that the book was already in the folder either....thus, the only option I saw, was to tell the client to again download the file....which of course puts the client and the server on the same note regarding who downloaded it....so, while this 'fixed' the issue with this book, I have several other books already downloaded as well, which my client no longer recognizes as downloaded....so....I don't feel this 'reset the client' thing is a fix for the problem....just causing more issues

**Comment by advplyr** [comment]
I don't see how this could happen. We'll first need to be able to reproduce the issue so if you can figure out a way to reproduce it that would be helpful

**Comment by brand421** [comment]
I'm having this issue as well. Media was downloaded on the same network as the server, with the same user being logged into the app the whole time.  Most recent version is installed on both phone and server.

**Comment by jhirning** [comment]
I'm also getting this message

**Comment by floatiepen** [comment]
I'm also getting this message

**Comment by jsk01** [comment]
Same after upgrading server to latest version.

**Comment by jnsimon** [comment]
I was seeing this as well with version 2.15.0 of the server and v0.9.76-beta of the android app. I force stopped the app and cleared the app storage. Logging back in to everything seems to have made the error go away.

**Comment by meyeaard** [comment]
First off, awesome work on this project, this is one of my most used apps and it's more polished than some commercial options.

I've been having this issue for some time now over multiple updates to the droid app and server. I've been making due, just dropped over to check on possible fixes after another server update hadn't fixed it.

I killed and cleared storage as @jnsimon commented, and for now it seems to work (only been 10 minutes or so really). But honestly it's not a fix I would want to tell my mom or wife they'd have to do as it does delete all client side data - and even through the audiobook files are in a separate folder it won't detect the media is already downloaded so I had to manually clean and re-download all the books my kids are I are in the middle of.

**Comment by neingeist** [comment]
I also encountered this (Pixel 8 on Android 14)

**Comment by Pwnz74** [comment]
Same. Updated to version v2.15.0. It was not before. I am not aware any changes that I made. Redownload of the audiobook did not helped it.

**Comment by ljlongwing** [comment]
Closing the loop on this one for anyone who lands here.

**Root cause** (confirmed in the server-side discussion at advplyr/audiobookshelf#3490): server **v2.13.0** (released 2024-08-31) removed the "old" pre-v2.4.0 user id. Local downloads made under that old id keep it in `localLibraryItem.serverUserId`, and `syncLocalMediaProgressForUser` filters progress by the matching id/connection, so sync silently stops and the "downloaded by a different user" warning shows permanently.

The app has a guard for exactly this — `isLocalMatchingUser` also checks `this.user.oldUserId` (added in #945) — but `User.toOldJSONForBrowser()` (what `/api/me` returns to the app) never includes `oldUserId`; it only lives in `user.extraData.oldUserId` server-side. So `this.user.oldUserId` is always `undefined` in the app and that fallback can't fire.

**Why it's largely moot now:** the stale ids stopped being issued around server v2.4.0 (~April 2023). A saved server connection refreshes its `userId` on every login, so after any login to a modern server all *new* downloads carry the current id. To still hit this today you'd need a book downloaded 3+ years ago, still on the device, still in use, never re-fetched. Replacing the download, re-adding the server connection, or a new phone each independently clears it.

It was a real problem for a cohort of users in late 2024; two years on it's a shrinking corner case. Fine to close unless fresh reports come in on a current setup.
